components/service-operator/apis/access/v1beta1/principal_types.go

```diff
@@ -91,9 +96,44 @@ func (s *Principal) GetStackTemplate() *cloudformation.Template {
 		"Type": "String",
 	}
 
+	template.Parameters[IAMOIDCProviderURLParameterName] = map[string]string{
+		"Type": "String",
+	}
+	template.Parameters[IAMOIDCProviderARNParameterName] = map[string]string{
+		"Type": "String",
+	}
+
+	// We have to build the assume role policy document as JSON and put it through Fn::Sub as we'll
+	// need one of the parameters (the OIDC provider URL) to go into a condition key, so we can't
+	// use Ref.
+	var err error
+	var policyDocJson []byte
+	if s.Spec.TrustServiceAccount == "" {
```

Comment:
Have we got a test for both of these cases? I may have missed them in the diff.

Reply:
Right now we do no testing on the AssumeRolePolicyDocument at all. We should probably change that.